mock_service/lib/dsl.py

```diff
@@ -98,9 +104,14 @@ def __resolve_value(action_context: ActionContext, value: Any) -> Any:
 
     if isinstance(value, str) and value.startswith("$"):
         if value == "$self":
-            return get_datapoint_value(
-                action_context.execution_context, action_context.datapoint.path
-            )
+            action_context.execution_context.client.connect()
```

Comment:
What's the reason for this change? Calling get_datapoint_value would save duplicate code.

Reply:
No, will change